commons-core/src/main/scala/com/avsystem/commons/serialization/GenCodec.scala

```diff
@@ -343,14 +356,43 @@ object GenCodec extends RecursiveAutoCodecs with TupleGenCodecs {
   implicit lazy val UuidCodec: GenCodec[UUID] =
     nullableSimple(i => UUID.fromString(i.readString()), (o, v) => o.writeString(v.toString))
 
-  private implicit class IterableOnceOps[A](private val coll: IterableOnce[A]) extends AnyVal {
-    def writeToList(lo: ListOutput)(implicit writer: GenCodec[A]): Unit =
-      coll.iterator.foreach(writer.write(lo.writeElement(), _))
+  private def declareSize(o: SequentialOutput, coll: BIterable[_]): Unit =
+    o.sizePolicy match {
+      case SizePolicy.Ignored =>
+      case SizePolicy.Optional =>
+        if (coll.isEmpty) {
```

Comment:
This case should be possible to handle in the second branch. Tests seem to confirm.